devito/tools/data_structures.py

```diff
@@ -29,18 +30,33 @@ def __init__(self, **kwargs):
         self.__dict__.update(kwargs)
 
 
-class EnrichedTuple(tuple):
+class EnrichedTuple(Pickable, tuple):
 
     """
     A tuple with an arbitrary number of additional attributes.
     """
 
     def __new__(cls, *items, getters=None, **kwargs):
+        # Check kwargs e.g for pickled input
```

Comment:
Assuming this stays (though I suspect the actual issue is somewhere else), the comments should be changed accordingly as pickling is another thing

pickling uses reconstructability 

but reconstructability is a more general concept

Reply:
The error is not somewhere else. This is reconstruced as a pure tuple and because of the constructor `__new__(cls, *items, getters=None, **kwargs)` and the fact that it modifies `__dict__` it is reconstucted as `__new__(cls, (*items,))`. So it misses all the extra nad it also reconstuct with the wrong items because `(2,3)` is then reconstructed as `((2, 3),)`